pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -972,4 +1017,41 @@ public void onUpdate(String path, Map<String, String> data, Stat stat) {
         return dynamicConfigurationMap;
     }
 
-}
+    /**
+     * Create pending topic and on completion it picks the next one until processes all topics in
+     * {@link #pendingTopicLoadingQueue}.<br/>
+     * It also tries to acquire {@link #topicLoadRequestSemaphore} so throttle down newly incoming topics and release
+     * permit if it was successful to acquire it.
+     */
+    private void createPendingLoadTopic() {
+        Pair<String, CompletableFuture<Topic>> pendingTopic = pendingTopicLoadingQueue.poll();
+        if (pendingTopic != null) {
+            final String topic = pendingTopic.getLeft();
+            try {
+                checkTopicNsOwnership(topic);
+                CompletableFuture<Topic> future = new CompletableFuture<>();
+                final Semaphore topicLoadSemaphore = topicLoadRequestSemaphore.get();
+                final boolean acquiredPermit = topicLoadSemaphore.tryAcquire();
+                createPersistentTopic(topic, future);
+                future.handle((persistentTopic, ex) -> {
+                    if (ex != null) {
+                        pendingTopic.getRight().completeExceptionally(ex);
+                    } else {
+                        pendingTopic.getRight().complete(persistentTopic);
+                    }
+                    // release permit and process next pending topic
+                    if (acquiredPermit) {
+                        topicLoadSemaphore.release();
+                    }
+                    createPendingLoadTopic();
+                    return null;
+                });
+            } catch (RuntimeException re) {
+                log.error("Failed to create pending topic {} {}", topic, re);
+                pendingTopic.getRight().completeExceptionally(re.getCause());
+                // process next pending topic
+                createPendingLoadTopic();
```

Comment:
`createPersistent` is async call so, we are making it recursive once future is completed otherwise due to throttling-limit=10 if queue has 1000 pending requests then all 1000 can go concurrently through `createPersistent(topic,future)` . So, we have to call it recursively only after receiving completed future. right?